honeybee_grasshopper_energy/src/HB Model to IDF.py

```diff
@@ -0,0 +1,128 @@
+            This will ensure that result files appear in their respective outputs.
+    
+    Returns:
+        report: Check here to see a report of the EnergyPlus run.
+        idf: The file path of the IDF file that has been generated on your machine.
+        sql: The file path of the SQL result file that has been generated on your
+            machine. This will be None unless run_ is set to True.
+        eio:  The file path of the EIO file that has been generated on your machine.
+            This file contains information about the sizes of all HVAC equipment
+            from the simulation.
+        rdd: The file path of the Result Data Dictionary (.rdd) file that is
+            generated after running the file through EnergyPlus.  This file
+            contains all possible outputs that can be requested from the EnergyPlus
+            model.  Use the Read Result Dictionary component to see what outputs
+            can be requested.
+        html: The HTML file path of the Summary Reports. Note that this will be None
+            unless the input _sim_par_ denotes that an HTML report is requested and
+            run_ is set to True.
+"""
+
+ghenv.Component.Name = "HB Model to IDF"
+ghenv.Component.NickName = 'ModelToIDF'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = '5 :: Simulate'
+ghenv.Component.AdditionalHelpFromDocStrings = "1"
+
+import os
+
+try:
+    from honeybee_energy.simulationparameter import SimulationParameter
+    from honeybee_energy.run import run_idf
+    from ladybug.designday import DDY
+    from ladybug.futil import write_to_file_by_name, nukedir
+    from ladybug_rhino.config import conversion_factor
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import Honeybee:\n\t{}'.format(e))
+
+
+if all_required_inputs(ghenv.Component) and _write:
+    # process the design days
+    if _ddy_file_ is None:
+        folder, epw_file_name = os.path.split(_epw_file)
+        ddy_file = os.path.join(folder, epw_file_name.replace('.epw', '.ddy'))
+        if os.path.isfile(ddy_file):
+            ddy_obj = DDY.from_ddy_file(ddy_file)
+            ddy_strs = [ddy.ep_style_string for ddy in ddy_obj.design_days if
+                        '99.6%' in ddy.name or '.4%' in ddy.name]
+        else:
+            raise ValueError('No _ddy_file_ has been input and no .ddy file was '
+                             'found next to the _epw_file.')
+    else:
+        ddy_obj = DDY.from_ddy_file(_ddy_file_)
+        ddy_strs = [ddy.ep_style_string for ddy in ddy_obj.design_days]
+    
+    # process the simulation parameters
+    if _sim_par_ is None:
+        _sim_par_ = SimulationParameter()
+        _sim_par_.output.add_zone_energy_use()
+    
+    # process the additional strings
+    add_str = '/n'.join(add_str_) if add_str_ is not None else ''
```

Comment:
Ok. I am glad that you seem to be in agreement that the additional strings should continue to exist.  Nearly every advanced user has told me that they make use of it and I think we can even add an E+ measure on the future OpenStudio component that adds the additional strings in before simulation.

And I think you're thinking of IDF parsing much more optimistically than we will be able to support.

There's no "idf object" in honeybee-energy and I purposefully haven't put together reverse translators from IDF files to honeybee-core geometry objects yet because (invariably) we won't be able to support the import of 95% of the objects that exist in the EnergyPlus IDD. Whatever we put together will be prone to errors if the IDF does not originate with Honeybee.  So, to manage this, I think we will need options about what should be imported (ie. geometry only or geometry + constructions + schedules + loads + ideal air systems, etc.). Also, the reverse translation process will likely be very lossy given that there are no ProgramTypes or ConstructionSets in IDFs and HVAC templates get commented out once you run an IDF through E+. I imagine most people might only want to import the geometry and then start again from scratch using ProgramTypes and ConstructionSets.

Reply:
You might be overthinking what I meant by parsing idf files. I didn't mean to re-create the objects from idf string as Python objects. My proposal is to just parse the idf to a dictionary and then add or update the objects if they already exist. This has come up a number of times that people want to use the additional string to modify one of the existing objects. At the same it might not be desired anymore now that in theory one should have access to all the model.properties.energy before passing them to this component. We can leave it as is for now and see if it will come up in the future.